docs/guides/typescript.md

```diff
@@ -0,0 +1,72 @@
+
+# TypeScript
+
+In this guide, we will go through the steps to get a Buidler project working with TypeScript. This means that you can write your Buidler config, tasks, scripts and tests in [TypeScript](https://www.typescriptlang.org/). For a general overview of using Buidler refer to the [Getting started guide](/guides/#getting-started).
+
+## Installing dependencies
+
+We will need to install the TypeScript packages to do this.
+
+In your terminal, run
+```npm i -D ts-node typescript```
```

Comment:
There are actually two cases:

- local installation: that's the one covered.

- global installation, you need to install ts-node and typescript globally

Reply:
See #351